client/guidestours/positioning.js

```diff
@@ -0,0 +1,61 @@
+const BULLSEYE_RADIUS = 6;
+const DIALOG_WIDTH = 410;
+const DIALOG_PADDING = 10;
+const MASTERBAR_HEIGHT = 48;
+
+const middle = ( a, b ) => Math.abs( b - a ) / 2;
+
+const wouldBeOffscreen = pos =>
+	( pos + DIALOG_PADDING + DIALOG_WIDTH ) > document.documentElement.clientWidth
+
+const ensureOnScreen = pos =>
+	Math.abs( pos - DIALOG_PADDING - DIALOG_WIDTH )
```

Comment:
I had to read the positioners to understand this better. Would a name like `fitToTheRight` / `fitFromRight` be easier for readers? Otherwise, keep it as is.